examples/prodlda.py

```diff
@@ -0,0 +1,332 @@
+        self._vocab_size = vocab_size
+        self._num_topics = num_topics
+        self._hidden = hidden
+        self._dropout_rate = dropout_rate
+
+    def __call__(self, inputs, is_training):
+        dropout_rate = self._dropout_rate if is_training else 0.0
+
+        h = jax.nn.softplus(hk.Linear(self._hidden)(inputs))
+        h = jax.nn.softplus(hk.Linear(self._hidden)(h))
+        h = hk.dropout(hk.next_rng_key(), dropout_rate, h)
+
+        # NB: here we set `create_scale=False` and `create_offset=False` to reduce
+        # the number of learning parameters
+        h_mu = hk.Linear(self._num_topics)(h)
+        logtheta_loc = hk.BatchNorm(
+            create_scale=False, create_offset=False, decay_rate=0.1
+        )(h_mu, is_training)
+
+        h_sigma = hk.Linear(self._num_topics)(h)
+        logtheta_logvar = hk.BatchNorm(
+            create_scale=False, create_offset=False, decay_rate=0.1
+        )(h_sigma, is_training)
+        logtheta_scale = jnp.exp(0.5 * logtheta_logvar)
+        return logtheta_loc, logtheta_scale
+
+
+class FlaxEncoder(nn.Module):
+    vocab_size: int
+    num_topics: int
+    hidden: int
+    dropout_rate: float
+
+    @nn.compact
+    def __call__(self, inputs, is_training):
+        h = nn.softplus(nn.Dense(self.hidden)(inputs))
+        h = nn.softplus(nn.Dense(self.hidden)(h))
+        h = nn.Dropout(self.dropout_rate, deterministic=not is_training)(h)
+
+        h_mu, h_sigma = nn.Dense(self.num_topics)(h), nn.Dense(self.num_topics)(h)
+
+        logtheta_loc = nn.BatchNorm(
+            use_bias=False, use_scale=False, use_running_average=not is_training
+        )(h_mu)
+        logtheta_logvar = nn.BatchNorm(
+            use_bias=False, use_scale=False, use_running_average=not is_training
+        )(h_sigma)
+        logtheta_scale = jnp.exp(0.5 * logtheta_logvar)
+        return logtheta_loc, logtheta_scale
+
+
+class HaikuDecoder:
+    def __init__(self, vocab_size, dropout_rate):
+        self._vocab_size = vocab_size
+        self._dropout_rate = dropout_rate
+
+    def __call__(self, inputs, is_training):
+        dropout_rate = self._dropout_rate if is_training else 0.0
+        h = hk.dropout(hk.next_rng_key(), dropout_rate, inputs)
+        h = hk.Linear(self._vocab_size, with_bias=False)(h)
+        h = hk.BatchNorm(create_scale=False, create_offset=False, decay_rate=0.1)(
+            h, is_training
+        )
+        return jax.nn.softmax(h)
+
+
+class FlaxDecoder(nn.Module):
+    vocab_size: int
+    dropout_rate: float
+
+    @nn.compact
+    def __call__(self, inputs, is_training):
+        h = nn.Dropout(self.dropout_rate, deterministic=not is_training)(inputs)
+        h = nn.Dense(self.vocab_size, use_bias=False)(h)
+        h = nn.BatchNorm(
+            use_bias=False, use_scale=False, use_running_average=not is_training
+        )(h)
+        return nn.softmax(h)
+
+
+def model(docs, hyperparams, is_training=False, use_haiku=False):
```

Comment:
I went with `nn_framework` rather than `nn_module` because I thought the latter might get confused with `nn.Module`? Probably doesn't matter but let me know if you would prefer `nn_module`.

Reply:
Yeah, agree. I like `nn_framework`. :D